### PR TITLE
Update synocli net

### LINKED ISSUE #3519: Arp-scan for synology?
There is a good utility in Linux called arp-scan. It shows computers in networks (ip, hostname, MAC). Is it possible to build spk with arp-scan? It might be very helpful.

### PR DESCRIPTION
**Pull request by hgy59**
_Motivation:_  Revise outdated synocli-net packages and add arp-scan
_Linked issues:_  #3742, #4196, closes #3519

### Checklist
- [x] Build rule `all-supported` completed successfully
- [x] Package upgrade completed successfully
- [x] New installation of package completed successfully

### Remarks
- add diyspk for some packages of synocli-net to verify updates before building the collection
- ~unfortunately with newer toolchains added, the rule `all-supported` does not use the TC 6.1 anymore but takes the newest TC available. Therefore I created the rule `all-general` to build TC 6.1 and ppc853x-5.2 version.~
- most of the contained packages depend on cross/openssl and my intention is to merge this to master (and not to publish) before the update of openssl to version 1.1.x.
- dependency of `Perl` is removed and `mosh` is moved to a dedicated package
- add `arp-scan` (includes three perl scripts that are not mandatory and therefore `SPK_DEPENDS=Perl` is omitted
- add `links` and remove dedicated package (moved to diyspk)

### CODE REVIEW

**Comment by hgy59** [comment]
rutorrent still fails to build (depending on updated screen) but I didn't find a fix so far.
So I propose to merge this PR to go further with openssl update.

**Comment by th0ma7** [comment]
Hi @hgy59 , a few questions for you:
> unfortunately with newer toolchains added, the rule all-supported does not use the TC 6.1 anymore but takes the newest TC available. Therefore I created the rule all-general to build TC 6.1 and ppc853x-5.2 version.

- Doesn't a `make setup` forces to use DSM6.1 thus building package as before?  Or should the `make all-supported` rule set be reviewed as well?
- Indeed the `openssl.cnf` is not installed anymore with the usage of `/etc/ssl` by default.  What was the reason to revert the change that you first did?  And should it apply to newer 1.1.1 version?
- For my own understanding what does the `diyspk` stands for?

**Comment by hgy59** [comment]
> * Doesn't a `make setup` forces to use DSM6.1 thus building package as before?  Or should the `make all-supported` rule set be reviewed as well?

`make setup` sets the default TC to DSM 6.1, but this is only for make arch-{arch} without specification of the TC version.
`make all-supported` does not regard the DSM version set with make setup - it takes the highest version available for each arch. I don't know why the 6.2 and 6.2.2 toolchains where added. We would need the exact version only for packages that require the kernel sources (afaik). Otherwise we (should) publish with the lowest version compatible with the actual version.

May be it is worth to remove my `all-general` target and adjust the `all-supported` target to work like this. ❓ 

_Another approach would be to regard the default TC version set with `make setup` - but with auto resolving the version when the exact version is not available. i.e. take the highest TC version of the arch if it is less than the default version - or - take the lowest TC version if versions greater than the default are available._

Other considerations:
- `make all-legacy` builds still for TC 4.3
- `make all-supported` should include the build for ppc853x-5.2 as this is a still supported version by synology.com.
- I suppose that with DSM 7 we will have to build for 6.1 and 7.0 in parallel, as some models will not get the update to DSM 7 (as DS210+ did not for DSM 6)

*****

> * Indeed the `openssl.cnf` is not installed anymore with the usage of `/etc/ssl` by default.  What was the reason to revert the change that you first did?  And should it apply to newer 1.1.1 version?

I removed the openssl.cnf because it is not used anymore and gives a warning due to missing the file. And I reverted it as this is not related to synocli-net and forces all the openssl dependent packages to be built by the github build action. (it sould be removed from the PLIST in a PR related to openssl)

*****
> 
> * For my own understanding what does the `diyspk` stands for?

This stands for do it your self packages, i.e. packages that can be built by developpers themself and that are not published (any more) to the SynoCommunity package repo. It was introduced with the creation of synocli-* package collections. Since the synocli collections contain a lot of packages it is very useful to have the single package diyspk versions to build and test updated packages that are part of synocli-*.

**Comment by th0ma7** [comment]
@hgy59 thnx for the answers, additional comments on it:
1. Agreed, now that I better understand the behaviour of `all-supported` I believe this really this should map to what we **do** support which should range from ppc853x-5.2 up to DSM-6.1 (optionnally new 6.2 only arches such as purley, geminilake, etc). This will need to be readjust later anyway with the arrival of DSM7.  With a similar mindset, `all-legacy` should be revisited as well but its not urgent for the moment (perhaps with DSM7). And yes 6.2 & 6.2.2 where added due to kernel related requirements.  Sadly 6.2.3 was never published by Synology which contains all the intel cpu security fixes that where backported.  This leads to module & kernel crashes at loading time when loading a 6.2.2 compiled module under a 6.2.3 running kernel... 
2. Good, I will ensure its there in the `openssl` PR (lack of time this week to look into it but I believe I already removed it)
3. Really interesting I didn't know... there may be some interest for sub `ffmpeg` packages like intel drivers for hardware acceleration for instance.  I'll keep that to mind.

**Review by th0ma7 — COMMENTED** [review]
Why removing sshfs? ... just re-read and figure you are not removing sshfs but rather moving it as diyspk instead of standalone spk but also keeing in into `synocli-net`, I'm all good witht hat.

But shouldn't the `make all-general` be superseeded with #4200 by now?

File: cross/arp-scan/patches/001-update-get-iab.patch
Comment (ymartin59) [inline cross/arp-scan/patches/001-update-get-iab.patch:2]: Please do not "hesitate" to submit such patch upstream to ease future maintenance/update

**Comment by ymartin59** [comment]
@hgy59 @th0ma7 I would like to avoid to create "some many" diyspk/ packages because it increases "maintenance cost" when there is "no need" or no obvious tips they will be used by anyone sooner-or-later.
First "diyspk" packages were introduced to keep previous "standalone package" for small applications still available (not to enforce users to change if able to build on their own), when introducing aggregation in "synocli" packages...

Notice that in previous discussions, I agree to extract "mosh" out of "synocli-net" as a standalone package spk/mosh because it is quite heavy and introduces dependency to Python. May you please move diyspk/mosh to spk/mosh and cut dependencies to make synocli-net far lighter.

File: mk/spksrc.common.mk
Comment (ymartin59) [inline mk/spksrc.common.mk:39]: IMO Conflicting with #4200. Please discard

File: mk/spksrc.spk.mk
Comment (ymartin59) [inline mk/spksrc.spk.mk:365]: IMO Conflicting with #4200. Please discard

**Review by ymartin59 — CHANGES_REQUESTED** [review]
I would also move spk/sshfs to diyspk/sshfs for "compatibility" only

**Comment by ymartin59** [comment]
According to this diyspk "migration" policy, it would make sense to remove them "one day". But again I would not create diyspk package for so small new tools.

**Comment by hgy59** [comment]
> According to this diyspk "migration" policy, it would make sense to remove them "one day". But again I would not create diyspk package for so small new tools.

I find it rather usefull to have such diyspk for the synocli-* collections and would like to keep them, as you can build and test single package updates without the need to builld and install the whole collection.

**Comment by ymartin59** [comment]
@hgy59 OK. For sure if they are used for testing, there is less chance to forget to update them. We will see... until then go on that way.
About mosh own package, would it be possible for you to take over what I started at #3820 (of course forget about Meson support which will be addressed later)
There is also a trouble to investigate with mosh (and probably tmux) concerned detached session: #3582 #3583
Again many thanks for your work here.

File: cross/arp-scan/patches/001-update-get-iab.patch
Comment (hgy59) [reply cross/arp-scan/patches/001-update-get-iab.patch]: @ymartin59 just found that this works with the current release

**Comment by hgy59** [comment]
I propose to rebase and merge only after #4155 is merged into master.

**Comment by hgy59** [comment]
@ymartin59 build fails due to error in openssl build (1.0.1u).
So I definitly wait for the merge of #4155 before finalizing this.

The error comes form installing openssl parts into /var/packages instead of using the install folder ../install/var/packages/synocli-net/target/...
I propose to forget openssl 1.0.1u and focus on 1.1.1.

**Comment by ymartin59** [comment]
@hgy59 The fix quite simple. I propose you grab it from my PR #4195 to confirm https://github.com/SynoCommunity/spksrc/blob/cbda607efcccafc1781cf4a80a14316012426e21/cross/openssl/Makefile#L59

**Comment by hgy59** [comment]
> 
> 
> @hgy59 The fix quite simple. I propose you grab it from my PR #4195 to confirm
> 
> https://github.com/SynoCommunity/spksrc/blob/cbda607efcccafc1781cf4a80a14316012426e21/cross/openssl/Makefile#L59

Yes this is the fix, but commiting would lead to merge conflicts - therefore I proposed to forget about openssl 1.0.1u.

**Comment by ymartin59** [comment]
@hgy59 No there will be no merge conflict with Git if change is stricly the exact same line, that is why it is worth trying.

**Pull request merged**
